chainntnfs/txnotifier.go

```diff
@@ -1227,6 +1176,93 @@ func (n *TxNotifier) DisconnectTip(blockHeight uint32) error {
 	return nil
 }
 
+// updateHints attempts to update the confirm and spend hints for all relevant
+// transactions and outpoints respectively. The height parameter is used to
+// determine which transactions and outpoints we should update based on whether
+// a new block is being connected/disconnected.
+//
+// NOTE: This must be called with the TxNotifier's lock held and after its
+// height has already been reflected by a block being connected/disconnected.
+func (n *TxNotifier) updateHints(height uint32) {
```

Comment:
nice refactor, really nice that both of these are being handled in one place. down the road i think it'd be worth considering updating both types of hints under a single db transaction, but we can cross that bridge when we get there

Reply:
Indeed - added a `TODO`.